### ISSUE TITLE
unsupported operand type(s)

### ISSUE THREAD

**Issue by m0ppers** [blueman-project/blueman#975]
blueman: 2.0.7
BlueZ: 5.50
Distribution: arch linux
Desktop environment: sway

* [ ] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

I cannot start blueman-manager. I am getting a typeerror:

```
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/dbus/connection.py", line 604, in msg_reply_handler
    reply_handler(*message.get_args_list(**get_args_opts))
  File "/usr/bin/blueman-manager", line 150, in on_bluez_name_owner_changed
    self.List.DisplayKnownDevices(autoselect=True)
  File "/usr/lib/python3.7/site-packages/blueman/gui/DeviceList.py", line 392, in DisplayKnownDevices
    self.device_add_event(Device(device))
  File "/usr/lib/python3.7/site-packages/blueman/gui/manager/ManagerDeviceList.py", line 207, in device_add_event
    self.AppendDevice(device)
  File "/usr/lib/python3.7/site-packages/blueman/gui/DeviceList.py", line 425, in AppendDevice
    self.add_device(device, True)
  File "/usr/lib/python3.7/site-packages/blueman/gui/DeviceList.py", line 332, in add_device
    self.row_setup_event(iter, device)
  File "/usr/lib/python3.7/site-packages/blueman/gui/manager/ManagerDeviceList.py", line 253, in row_setup_event
    description = get_major_class(device.Class).capitalize()
  File "/usr/lib/python3.7/site-packages/blueman/DeviceClass.py", line 240, in get_major_class
    index = (klass >> 8) & 0x1F
TypeError: unsupported operand type(s) for >>: 'str' and 'int'

```

I have put a stupid fix in there (which is probably not the right fix) and can now start it:

https://github.com/blueman-project/blueman/blob/master/blueman/DeviceClass.py#L234
```
    if klass == 'unknown':
        return 'unknown'

```

Probably someone knowing what they are doing should be able to create a real fix :D Not at all urgent as my hotfix helps me so far

**Comment by infirit** [comment]
Thanks for the report. Can you try below please.

Also I am interested in what the Class id is so can you post the log of blueman-manager, run it from a terminal and post it into a github gist :smile: .

```patch
diff --git a/blueman/gui/manager/ManagerDeviceList.py b/blueman/gui/manager/ManagerDeviceList.py
index b5ac8686..51af1c0c 100644
--- a/blueman/gui/manager/ManagerDeviceList.py
+++ b/blueman/gui/manager/ManagerDeviceList.py
@@ -250,7 +250,7 @@ class ManagerDeviceList(DeviceList):
             description = gatt_appearance_to_name(appearance)
         else:
             icon = get_icon(device.Icon, 48, "blueman")
-            description = get_major_class(device.Class).capitalize()
+            description = klass.capitalize()
 
         name = device.Alias
         address = device.Address
```

**Comment by m0ppers** [comment]
that works as well :)

here is the gist:

(Added "Klass: " and "Device.class:" there)

https://gist.github.com/m0ppers/d7d7d0ac7a149adb4d433e16809ccc83

**Comment by infirit** [comment]
Thanks. This was backported from master and I introduced this bug :sob:.

**Comment by abhixec** [comment]
Thanks, that fixed the issue for me too on Archlinux.

**Comment by Mange** [comment]
Also worked for me. Is there any chance to get this shipped soon so I don't have to manually apply this patch to all my machines?

blueman-manager 2.0.7

**Comment by cschramm** [comment]
What do mean, @Mange? We released version 2.0.8 a month ago, so if you're talking about the Arch package that's on Arch. https://www.archlinux.org/packages/community/x86_64/blueman/ it's flagged out-of-date but there seems to be no issue or any other information on an update to 2.0.8.

**Comment by Mange** [comment]
I apologize. :bowing_man: 

Since I couldn't get a newer version on Arch, and this issue was still open, I wrongly assumed that no fix had been released. I should have checked more closely, and will do so the next time.
